tools/render-test/cuda/cuda-compute-util.cpp

```diff
@@ -58,6 +58,52 @@ class CUDAResource : public RefObject
     void* m_cudaMemory;
 };
 
+class CUDATextureResource : public RefObject
```

Comment:
I probably need a refresher on this, but I keep wondering why CUDA (and the CPU path) aren't being handled as `Renderer`s that only support compute shaders for now.

I guess this all has to do with the binding differences, and once we've fixed up the issues that require by-name binding and related hackery to make CPU binding work in in the tests we can reconsider making CPU + CUDA just be two more `Renderer` implementations.

Reply:
Well ... that is what you advocated for, actually on both, although probably because on CPU that was a more straight forward argument. And yes binding was arguably a big part of that. And as it turns out CPU and CUDA binding as used here are quite similar.

We might also categorize say CPU/CUDA differently as being only 'compute' which might make things simpler.  

But it could be implemented that way. Or we could wrap up an abstraction that made doing this simpler.